webserver.cpp

```diff
@@ -28,7 +29,7 @@ WebServer::~WebServer() {
 
 void WebServer::begin() {
 
-    for (auto it = SmartDevice::alldevices.begin(); it != SmartDevice::alldevices.end(); ++it) {
+    for (auto it = SmartManager::devices.begin(); it != SmartManager::devices.end(); ++it) {
```

Comment:
inject SmartManager's instance into the WebServer and use that instance to iterate thru: `smartManager->devices` better yet, you can provide a method that provide an iterator to your devices